quiltctl/command/init/initializer.js

```
@@ -0,0 +1,213 @@
+  */
+function createFileFromTemplate(templateLocation, context, destination) {
+  if (!fs.existsSync(templateLocation)) {
+    throw new Error(`failed to create file at ${destination}.` +
+      ` No template in ${templateLocation}.`);
+  }
+  const rawTemplate = fs.readFileSync(templateLocation, 'utf8');
+  const template = handlebars.compile(rawTemplate);
+  fs.writeFileSync(destination, template(context));
+}
+
+/**
+  * Create the infrastructure file to be used with baseInfrastructure().
+  *
+  * @param {string} infraName The user chosen name of this infrastructure.
+  * @param {Object} config The context to fill into the infrastructure template.
+  * @return {void} Returns nothing.
+  */
+function writeInfraFile(infraName, config) {
+  const outputPath = util.infraPath(infraName);
+  createFileFromTemplate(infraTemplateFile, config, outputPath);
+  util.logExceptOnTest(`Created infrastructure in ${outputPath}`);
+}
+
+/**
+  * Given a file path, recursively create all directories in the path.
+  * This function assumes that the passed in path is to a file, not a directory.
+  *
+  * @param {string} filePath The complete file path for which to create
+  *  directories.
+  */
+function makeDirPath(filePath) {
+  const lastSlash = filePath.lastIndexOf('/');
+  const directories = filePath.slice(0, lastSlash);
+  const fullPath = path.join(os.homedir(), directories);
+
+  fsExtra.mkdirp(fullPath, (err) => {
+    if (err) {
+      throw new Error(
+        `failed to create credentials file in ${fullPath}: ${err}`);
+    }
+  });
+}
+
+/**
+  * Create an SSH key pair in the default Quilt SSH key location.
+  *
+  * @return {boolean} True on success, false on failure.
+  */
+function sshGenerateKeyPair() {
+  try {
+    execSync(
+      `ssh-keygen -t rsa -b 2048 -f ${consts.quiltSshKeyLocationPrivate}` +
+      ` -N '' -C ''`,
+      {stdio: 'ignore'});
+    return true;
+  } catch (err) {
+    return false;
```

I'm slightly worried that this swallows the error.  I wonder if it makes sense to pull the caller's log message here, so you can include the error returned by ssh-keygen in it's error message.  Or alternatively I suppose you could return the err.  Not quite sure what's idiomatic.